dbclient/WorkspaceClient.py

```diff
@@ -260,34 +271,35 @@ def download_notebooks(self, ws_log_file='user_workspace.log', ws_dir='artifacts
         with open(ws_log, "r") as fp:
             # notebook log metadata file now contains object_id to help w/ ACL exports
             # pull the path from the data to download the individual notebook contents
-            for notebook_data in fp:
-                notebook_path = json.loads(notebook_data).get('path', None).rstrip('\n')
-                dl_resp = self.download_notebook_helper(notebook_path, checkpoint_notebook_set, notebook_error_logger,
-                                                        export_dir=self.get_export_dir() + ws_dir)
-                if 'error' not in dl_resp:
-                    num_notebooks += 1
+            with ThreadPoolExecutor(max_workers=num_parallel) as executor:
+                futures = {executor.submit(self.download_notebook_helper, notebook_data, checkpoint_notebook_set, notebook_error_logger, self.get_export_dir() + ws_dir): notebook_data for notebook_data in fp}
```

Comment:
more python like would be 
`[executor.submit(self.download_notebook_helper, notebook_data, checkpoint_notebook_set, notebook_error_logger, self.get_export_dir() + ws_dir) for notebook_data in fp]`

Reply:
√